wkcuber/api/View.py

```diff
@@ -156,6 +161,40 @@ def _check_chunk_size(self, chunk_size):
                 f"The passed parameter 'chunk_size' {chunk_size} must be a multiple of (32, 32, 32)."
             )
 
+    def _handle_compressed_write(self, absolute_offset, data):
+        if (
+            self.header.block_type == wkw.Header.BLOCK_TYPE_LZ4
+            or self.header.block_type == wkw.Header.BLOCK_TYPE_LZ4HC
+        ):
+            # data is compressed
+
+            # calculate aligned bounding box
+            file_bb = np.full(3, self.header.file_len * self.header.block_len)
+            margin_to_top_left = np.array(absolute_offset) % file_bb
+            aligned_offset = np.array(absolute_offset) - margin_to_top_left
+            bottom_right = np.array(absolute_offset) + np.array(list(data.shape)[-3:])
+            margin_to_bottom_right = file_bb - (bottom_right % file_bb)
+            aligned_bottom_right = bottom_right + margin_to_bottom_right
+            aligned_shape = aligned_bottom_right - aligned_offset
+
+            if tuple(aligned_offset) != tuple(absolute_offset) or tuple(
+                aligned_shape
+            ) != tuple(list(data.shape)[-3:]):
+                # the data is not aligned
+                # read the aligned bounding box
+                aligned_data = self.read(offset=aligned_offset, size=aligned_shape)
```

Comment:
This also works, when reading and writing multiple chunks, right? They only have to be aligned, I guess.

Reply:
Very good point. It failed if the view (and therefore also the chunks) had the parameter `is_bounded` set to `True`. 
If the bounding box of the chunks was not aligned, `read(offset=aligned_offset, size=aligned_shape)` would try to access data outside of the bounding box (which is not allowed if `is_bounded=True`).

Reply:
I think we could run into concurrency problems: `View.for_each_chunk(...)` executes a function for each chunk. Let's say that the chunks are not aligned with the compressed files. Now each chunk reads the compressed file (not only its own bounding box) and then writes the compressed data.
Maybe we shouldn't allow this for chunks?!